src/ActionMenu.tsx

```diff
@@ -0,0 +1,76 @@
+import {List, ListPropsBase, GroupedListProps} from './ActionList/List'
+import {Item, ItemProps} from './ActionList/Item'
+import {Divider} from './ActionList/Divider'
+import Button, {ButtonProps} from './Button'
+import React, {useCallback, useRef, useState} from 'react'
+import Overlay from './Overlay'
+import getRandomValues from 'get-random-values'
+
+export interface ActionMenuProps extends ListPropsBase, GroupedListProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  renderAnchor?: (props: any) => JSX.Element
+  triggerContent?: React.ReactNode
+  renderItem?: (props: ItemProps) => JSX.Element
+  onActivate?: (props: ItemProps) => void
```

Comment:
onActivate is passed in and will trigger when a menu item is clicked or selected with the keyboard keys. This will pass ItemProps into the callback so the action can use the selected item to respond to the click or keyboard event.

This onActivate method might be more useful to users if we allowed a `meta` key of item props, where users could store actionable information about an item that could be used by onActivate to perform the items task/route to the proper place.

Reply:
Out of curiosity: Does (a `Menu`-level) `onActivate` support any use case that couldn’t be handled by (`Item`-level) `onClick` handlers?

Reply:
I think part of what this is trying to make a bit nicer is handling both click and keyboard functionality in one interface. I think to do the same thing with item level handlers would end up being a bit messy.